website/docs/language/tests/index.mdx

```diff
@@ -766,3 +826,191 @@ There are instances when Terraform does not execute a custom condition during th
 Other kinds of failure _besides_ the specified expected failures in the checkable object still result in the overall test failing. For example, a variable that expects a boolean value as input fails the surrounding test if Terraform provides the wrong kind of value, even if that variable is included in an `expect_failures` attribute.
 
 The `expect_failures` attribute is included to allow authors to test their configuration and any logic defined within. A type mismatch, as in the previous example, is not something Terraform authors should have to worry about testing as Terraform itself will handle enforce type constraints. As such, you can only `expect_failures` in custom conditions.
+
+## Parallel execution
+
+By default, Terraform executes `run` blocks sequentially. However, you can set the `parallel` attribute to `true` in the file's `test` block or in individual `run` blocks to enable parallel execution.
+
+When `parallel = true` for a `run` block, Terraform will attempt to parallelize such block wherever possible. Two or more `run` blocks can be executed in parallel if:
+
+- They do not reference outputs from each other.
+- They do not share the same state file. The state file is determined by the state key or the module source (when the state key is not set). Therefore, `run` blocks that share the same module configuration must specify different state keys to be executed in parallel.
+- They both have the `parallel` attribute set to `true`.
+- All other `run` blocks that they depend on have completed.
```

Comment:
This is when they run in parallel rather than if they can so I think the last bullet can be removed here?